nmdc/messages.go

```diff
@@ -1106,7 +1110,14 @@ func (m *Search) unmarshalString(data []byte) error {
 			if err != nil {
 				return err
 			}
-			m.Pattern = strings.Replace(string(str), "$", " ", len(str))
+			pattern := string(str)
+			if m.DataType == DataTypeTTH {
+				if !strings.HasPrefix(pattern, "TTH:") {
+					return fmt.Errorf("invalid TTH search")
+				}
+				pattern = strings.TrimPrefix(pattern, "TTH:")
```

Comment:
If we are going to parse this, it's better to add another field to the structure: `TTH *tiger.Hash`